velox/functions/lib/window/NthValue.cpp

```diff
@@ -101,6 +90,29 @@ class NthValueFunction : public exec::WindowFunction {
   }
 
  private:
+  // Helper function to update the constant offset value.
+  void updateConstantOffset(
+      const exec::WindowFunctionArg& offsetArg,
+      const TypePtr& offsetType) {
+    if (offsetType->isInteger()) {
+      constantOffset_ =
```

Comment:
At least for constantOffset_ we must stick just int64_t. Keeps it simple.
May be for vector offsets having 2 different sizes is a space optimization.

Reply:
@karteekmurthys Why do we have to use 64-bit integer for constant offset. Is it actually possible to have an offset that > max 32-bit integer?

Reply:
The reason 32-bit offset was added was to support Spark NthValue. I suggested to remove to reduce `if-else` checks at runtime. The Signature for PRESTO requires BIGINT while SPARK only INTEGER().

Reply:
Yes, I understand that Presto use BIGINT to represent offset, but can this value actually be bigger than max 32-bit integer. I assume not.

Reply:
@mbasmanova : Do we want to make that assumption ? Our code uses vector_size_t overall for partition sizes, but we could have massive partitions also. wdyt ?